concrete-csprng/src/counter/mod.rs

```diff
@@ -256,64 +279,71 @@ impl<G: AesBatchedGenerator> AesCtrGenerator<G> {
         &self.state
     }
 
-    /// Returns the bound of the generator if any.
-    pub fn get_bound(&self) -> Option<&State> {
-        self.bound.as_ref()
-    }
-
-    /// Returns whether the generator is bounded.
-    pub fn is_bounded(&self) -> bool {
-        self.bound.is_some()
+    /// Returns the bound of the generator.
+    pub fn get_bound(&self) -> &State {
+        &self.bound
     }
 
-    /// Returns the number of remaining bytes if bounded
-    pub fn remaining_bytes(&self) -> Option<usize> {
-        self.bound.as_ref().map(|bound| {
-            let res = ((bound.aes_ctr.0 - self.state.aes_ctr.0) as i128) * 16
-                + (bound.byte_ctr.0 as i128 - self.state.byte_ctr.0 as i128);
-            res as usize
-        })
+    /// Returns the number of remaining bytes
+    pub fn remaining_bytes(&self) -> u128 {
+        let step = self.step.0 as u128;
+
```

Comment:
I think both the bound and the state should be normalized, otherwise, we are not sure that the bound aes_ctr is always greater than the state aes_ctr.